frontend/model/contracts/group.js

```diff
@@ -995,10 +995,14 @@ sbp('chelonia/defineContract', {
         if (Object.keys(invite.responses).length === invite.quantity) {
           invite.status = INVITE_STATUS.USED
         }
-        // TODO: ensure `meta.username` is unique for the lifetime of the username
-        //       since we are making it possible for the same username to leave and
-        //       rejoin the group. All of their past posts will be re-associated with
-        //       them upon re-joining.
+        // NOTE: ensure `meta.username` is unique for the lifetime of the username
+        //       since we are making it possible for the same username to leave and rejoin the group.
+        //       All of their past posts will be re-associated with them upon re-joining.
+        const profile = state.profiles[meta.username]
+        if (profile && profile.status !== PROFILE_STATUS.REMOVED) {
+          console.error(`inviteAccept: ${meta.username} is currently active member in the group`)
+          throw new Errors.GIErrorIgnoreAndBan(`${meta.username} is already joined and active in the group`)
+        }
```

Comment:
Nice job @Silver-IT considering this. However, I'm worried that this might get some members banned unintentionally if we make a mistake in the code...

For example, we could add additional statuses and forget to update other code, or they could be in `PENDING` status... or something else.

The thing is, once the `e2e-protocol` is implemented properly, that should take care of ensuring that all messages are sent by who they claim to be sent by. So do you think you can remove these updates?

Reply:
OK, I see that this change is related to #1579. And indeed you are correct that calling `inviteAccept` twice is what resulted in the profile being reset (because the invite link was used twice).

I was able to track this down by using @snowteamer's `chel eventsSince` PR: https://github.com/okTurtles/chel/pull/9

For reference (for myself and others), this is the command on how to show all of the actions from the user `andrea` in the group:

```
chel eventsSince --limit 1000 groupincome.db 21XWnNWZzYYPfezAq9HN85b4by9TWt3o5GPUVzw9baSVjeQ49w 21XWnNWZzYYPfezAq9HN85b4by9TWt3o5GPUVzw9baSVjeQ49w | jq '.[] | select(.op[0] == "ae") | .op[1] | fromjson | select(.meta.username == "andrea")'
```

So, we still need to change this code because we don't want to ban Andrea for clicking the invite link twice. :P

Instead, if we are an existing user in the group, can we simply skip running the next line?

Reply:
@taoeffect, I have just removed that change in `group.js` contract file.
That's because it can't be reached since I prehandled this issue in `Join.vue` page.